api/features/views.py

```diff
@@ -851,48 +853,58 @@ def _additional_filters(self) -> Q:
 
         return filters
 
-    def _get_flags_from_cache(self, environment):  # type: ignore[no-untyped-def]
+    def _get_flags_from_cache(
+        self,
+        environment: Environment,
+        from_replica: bool = False,
+    ) -> list[typing.Any]:
+        data: list[typing.Any]
         data = flags_cache.get(environment.api_key)
         if not data:
             data = self.get_serializer(
                 get_environment_flags_list(
                     environment=environment,
                     additional_filters=self._additional_filters,
+                    from_replica=from_replica,
                 ),
                 many=True,
             ).data
             flags_cache.set(environment.api_key, data, settings.CACHE_FLAGS_SECONDS)
 
         return data
 
-    def _get_flags_response_with_identifier(self, request, identifier):  # type: ignore[no-untyped-def]
-        identity, _ = Identity.objects.get_or_create(
+    def _get_flags_response_with_identifier(
+        self, request: Request, identifier: str
+    ) -> Response:
+        identity, is_new_identity = Identity.objects.get_or_create(
             identifier=identifier, environment=request.environment
         )
 
-        kwargs = {
-            "identity": identity,
-            "environment": request.environment,
-            "feature_segment": None,
-        }
+        use_replica = not is_new_identity
+        feature_states = identity.get_all_feature_states(from_replica=use_replica)
 
         if "feature" in request.GET:
-            kwargs["feature__name__iexact"] = request.GET["feature"]
-            try:
-                feature_state = identity.get_all_feature_states().get(  # type: ignore[attr-defined]
-                    feature__name__iexact=kwargs["feature__name__iexact"],
-                )
-            except FeatureState.DoesNotExist:
+            is_case_insensitive = (
+                request.environment.project.only_allow_lower_case_feature_names
+            )
+            for feature_state in feature_states:
```

Comment:
Rather than this loop, can't we just use the `additional_filters` argument to the `get_all_feature_states` above? I don't really want to prioritise this code too much since it isn't actively used, but I don't quite understand the need for the loop / regex matching.